src/rgw/rgw_rest_role.cc

```diff
@@ -112,6 +112,16 @@ int RGWRestRole::verify_permission(optional_yield y)
   return 0;
 }
 
+int RGWRestRole::init_processing(optional_yield y)
+{
+  op_ret = verify_permission(y);
```

Comment:
don't call `verify_permission()` directly here, just move the code that initializes `_role`:
```c++
  string role_name = s->info.args.get("RoleName");
  std::unique_ptr<rgw::sal::RGWRole> role = driver->get_role(role_name,
							    s->user->get_tenant());
  if (int ret = role->get(s, y); ret < 0) {
    if (ret == -ENOENT) {
      return -ERR_NO_ROLE_FOUND;
    }
    return ret;
  }
  _role = std::move(role);
  return 0;
```

Reply:
```
int RGWRestRole::init_processing(optional_yield y)
{
    string role_name = s->info.args.get("RoleName");
    std::unique_ptr<rgw::sal::RGWRole> role = driver->get_role(role_name,
                                                               s->user->get_tenant());
    if (int ret = role->get(s, y); ret < 0)
    {
        if (ret == -ENOENT)
        {
            return -ERR_NO_ROLE_FOUND;
        }
        return ret;
    }
    _role = std::move(role);
    return 0;
}
```
added the method to initialise role inside the `init_processing`